compiler/rustc_mir_transform/src/promote_consts.rs

```diff
@@ -175,6 +176,11 @@ fn collect_temps_and_candidates<'tcx>(
 struct Validator<'a, 'tcx> {
     ccx: &'a ConstCx<'a, 'tcx>,
     temps: &'a mut IndexSlice<Local, TempState>,
+    /// For backwards compatibility, we are promoting function calls in `const`/`static`
+    /// initializers. But we want to avoid evaluating code that otherwise would not have been
+    /// evaluated, so we only do thos in basic blocks that are guaranteed to evaluate. Here we cache
```

Comment:
```suggestion
    /// evaluated, so we only evaluate constants from basic blocks that are guaranteed to be evaluated. Here we cache
```

Reply:
That doesn't sound quite right, this is about what we promote, not what we evaluate. I'll try to clarify the comment.